lib/parse.js

```diff
@@ -1,4 +1,25 @@
+function getType(v) {
+  if (v === undefined || v === null || typeof v === 'function') {
+    return 'any';
+  } else if (typeof v !== 'object') {
+    return typeof v;
+  } else {
+    return Array.isArray(v) ? 'array' : (Buffer.isBuffer(v) ? 'buffer' : 'object');
+  }
+};
+
 function formatBuffer(b) {
+  let type = getType(b);
+
+  if (type === 'object') {
+    Object.keys(b).map((k) => {
+      b[k] = formatBuffer(b[k]);
+    });
+    return b;
```

Comment:
The above should be (for the sake of copying data);

return Object.keys(b).reduce(function (obj, key) {
  obj[key] = formatBuffer(b[key]);
  return obj;
}, {});